apps/web/src/components/AssessmentResultsTable/AssessmentResultsTable.tsx

```diff
@@ -0,0 +1,190 @@
+import { buildColumn, columnHeader, columnStatus } from "./utils";
+import { AssessmentStepResult } from "./types";
+
+const columnHelper = createColumnHelper<AssessmentStepResult>();
+
+interface AssessmentResultsTableProps {
+  poolCandidate: PoolCandidate;
+}
+
+const AssessmentResultsTable = ({
+  poolCandidate,
+}: AssessmentResultsTableProps) => {
+  const intl = useIntl();
+
+  // Get assessment steps from pool
+  const assessmentSteps: Array<AssessmentStep> = unpackMaybes(
+    poolCandidate?.pool?.assessmentSteps,
+  );
+
+  // Get all pool skills from assessment steps and remove duplicates
+  const poolSkills: Array<PoolSkill> = unpackMaybes(
+    poolCandidate?.pool?.poolSkills,
+  );
+
+  // create assessment step results from pool skills
+  const assessmentResults: Array<AssessmentResult> = unpackMaybes(
+    poolCandidate?.assessmentResults,
+  );
+
+  // Create data for table containing pool skill with matching results
+  const assessmentStepsResults: Array<AssessmentStepResult> = poolSkills.map(
+    (poolSkill) => {
+      const matchingAssessmentResults = assessmentResults.filter(
+        (result) => result.poolSkill?.id === poolSkill.id,
+      );
+      return {
+        poolSkill,
+        assessmentResults: matchingAssessmentResults,
+      };
+    },
+  );
+
+  const educationResults = assessmentSteps
+    .find(
+      (assessmentStep) =>
+        assessmentStep.type === AssessmentStepType.ApplicationScreening,
+    )
+    ?.assessmentResults?.filter(notEmpty)
+    .filter(
+      (assessmentResult) =>
+        assessmentResult.assessmentResultType ===
+        AssessmentResultType.Education,
+    );
+
+  // Create the education requirement assessment step result
+  const educationStepResult: AssessmentStepResult = {
+    poolSkill: undefined,
+    assessmentResults: educationResults ?? [],
+  };
+
+  const getColumns = () => {
+    // Sort the pools assessment steps then build columns for the poolCandidates assessment results
+    const sortedAssessmentSteps = assessmentSteps.sort((a, b) => {
+      if (a.sortOrder && b.sortOrder) {
+        return a.sortOrder - b.sortOrder;
+      }
+
+      if (a.title && b.title) {
+        return Intl.Collator(intl.locale).compare(
+          getLocalizedName(a.title, intl),
+          getLocalizedName(b.title, intl),
+        );
+      }
+
+      return 0;
+    });
+    const columns = sortedAssessmentSteps.reduce(
+      (
+        accumulator: ColumnDef<AssessmentStepResult>[],
+        assessmentStep: AssessmentStep,
+      ) => {
+        const type = assessmentStep.type ?? null;
+        const id =
+          getAssessmentStepType(type ?? "unknownType").id ??
+          uniqueId("results-table-column");
+        const status = columnStatus(assessmentStep, assessmentResults);
+        const header = columnHeader(
+          intl.formatMessage(getAssessmentStepType(type ?? "unknownType")),
+          status,
+          type,
+          intl,
+        );
```

Comment:
Ya I noticed the same with the mobile view! Not sure why it's happening since column `header` says it [accepts](https://tanstack.com/table/latest/docs/guide/column-defs#column-helpers) `JSX.Element` type.

Reply:
Any progress on this one?

Reply:
Okay, I think this stems from this part of the responsive table component, the one that generates the toggle dialog. It is expecting strings for the header. I don't think this can be resolved within this PR as it will require updating the table component to adjust labels for when column headers are not strings. 

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/9faf5f3c-fed3-49cb-9417-cd906c5f4e81)

Reply:
I was able to fix the column dialog, but the mobile view has some more layers to it like you said. I'm okay with moving it to another PR though.

The column dialog is fixed by [faf2b1a](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9324/commits/faf2b1a0d9c5e7c885bf974c631152dff88274fd)

Reply:
I'll hide the mobile view header until we can get that next issue done

Reply:
I'll hide the mobile view header until we can get that next issue done

Reply:
Please spin off an issue that quickly summarizes what to reverse or to fix

Reply:
When toggled off it goes back to the original

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/dec8f2c9-8249-42e7-9528-363b59f3d3f4)

Reply:
I worked with Eric and we decided to just override the column header with a new meta prop. Fixed by [8e2a397](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9324/commits/8e2a397a65fed464544602b0f43a25011efbb485)